wherehows-web/app/components/dataset-compliance.ts

```diff
@@ -902,6 +970,56 @@ export default class DatasetCompliance extends Component {
   }
 
   actions: IDatasetComplianceActions = {
+    /**
+     * Toggle the visibility of the guided compliance edit view vs the advanced edit view modes
+     * @param {boolean} toggle flag ,if true, show guided edit mode, otherwise, advanced
+     */
+    onShowGuidedEditMode(this: DatasetCompliance, toggle: boolean): void {
+      const isShowingGuidedEditMode = set(this, 'showGuidedComplianceEditMode', toggle);
+
+      if (!isShowingGuidedEditMode) {
+        this.actions.onManualComplianceUpdate.call(this, get(this, 'jsonComplianceEntities'));
+      }
+    },
+
+    /**
+     * Handles updating the list of compliance entities when a user manually enters values
+     * for the compliance entity metadata
+     * @param {string} updatedEntities json string of entities
+     */
+    onManualComplianceUpdate(this: DatasetCompliance, updatedEntities: string): void {
+      try {
+        // check if the string is parseable as a JSON object
+        const entities = JSON.parse(updatedEntities);
+        const metadataObject = {
+          complianceEntities: entities
+        };
+        const isValid = validateMetadataObject(metadataObject, complianceEntitiesTaxonomy);
+
+        set(this, 'isManualApplyDisabled', !isValid);
+
+        if (isValid) {
+          set(this, 'manuallyEnteredComplianceEntities', metadataObject);
+        }
+      } catch {
+        set(this, 'isManualApplyDisabled', true);
```

Comment:
Would it be helpful to catch what kind of error we are getting here in order to help the user understand why this is disabled? Like JSON.parse returning an error would be a different cue than `validateMetadataObject()` returning an error I'd assume

Reply:
👍 yea, that's an enhancement to be worked on, for the first case, right now if the JSON is invalid, the editor shows an error indicator on the line where the issue occurred.
For the other case, `validateMetadataObject` is a boolean operation atm, the enhancement would be to fix this by capturing an propagating the issue that caused the validation to fail and showing that to the user